### PR TITLE
fix handling of piped completions near start of doc

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/13611.

### Approach

We have some internal logic that attempts to find the lead "data" object from a piped expression. However, this lookup was short-circuiting if the head of the pipe happened to be the first token in the document, but didn't exist at the very first row / column.

### Automated Tests

Included.

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/13611.

### Documentation

N/A

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/tests/automation/testthat/test-automation-completions.R [hunk 1]
```diff
@@ -28,3 +28,26 @@ test_that("autocompletion doesn't trigger active bindings", {
...
+   
+   remote$documentOpen(".R", code)
+   editor <- remote$editorGetInstance()
+   editor$gotoLine(2L, 26L)
+   remote$keyboardExecute("<Tab>")
+   popupEl <- remote$jsObjectViaSelector("#rstudio_popup_completions")
+   popupText <- popupEl$innerText
+   expect_true(grepl("trim =",  popupText, fixed = TRUE))
+   expect_true(grepl("na.rm =", popupText, fixed = TRUE))
+   expect_true(grepl("... =",   popupText, fixed = TRUE))
```
Comment (olivroy) [inline src/cpp/tests/automation/testthat/test-automation-completions.R:48]: Hi @kevinushey! Thanks for taking a look at this issue! However I am not sure this is the correct fix

```suggestion
   expect_true(grepl("x =",  popupText, fixed = TRUE))
   expect_true(grepl("... =",   popupText, fixed = TRUE))
```

And I would expect this to be true if not preceded by a single blank line.

See my comment https://github.com/rstudio/rstudio/issues/13611#issuecomment-1711857799

Comment (olivroy) [reply src/cpp/tests/automation/testthat/test-automation-completions.R:48]: See also https://github.com/davidhodge931/ggblanket/issues/661 im case that's clearer

Comment (olivroy) [inline src/cpp/tests/automation/testthat/test-automation-completions.R:48]:
```suggestion
   code <- .rs.heredoc('
      
      mtcars |> mutate(x = mean())
      mtcars |> mutate(x = mean())
   ')
   
   remote$documentOpen(".R", code)
   editor <- remote$editorGetInstance()
   editor$gotoLine(2L, 26L)
   remote$keyboardExecute("<Tab>")
   popupEl <- remote$jsObjectViaSelector("#rstudio_popup_completions")
   popupText <- popupEl$innerText
   expect_true(grepl("x =",  popupText, fixed = TRUE))
   expect_true(grepl("... =",   popupText, fixed = TRUE))

   editor$gotoLine(3L, 26L)
   remote$keyboardExecute("<Tab>")
   popupEl <- remote$jsObjectViaSelector("#rstudio_popup_completions")
   popupText <- popupEl$innerText
   expect_true(grepl("x =",  popupText, fixed = TRUE))
   expect_true(grepl("... =",   popupText, fixed = TRUE))
```

The test should possibly look like this

**Comment by kevinushey** [comment]
Thanks @olivroy; I think you're describing a related but separate issue. Either way, I'll take a look.

**Comment by olivroy** [comment]
Oh! You are right! I have updated my example test to test for the request at position 32, for 26, you are fixing the fact that we don't get column completion here:
(current behavior)
image

Apologies, I should have been more careful. I got too excited when I saw you were addressing the issue yesterday!

**Review by gtritchie — APPROVED** [review]
LGTM!

**Pull request merged**